.github/workflows/check-in-tree-build.yml

```diff
@@ -0,0 +1,175 @@
+# This workflow is intended to check if PR doesn't introduce any regressions
+# into in-tree build in various configurations.
+#
+# Documentation for GitHub Actions:
+# [workflow-syntax]: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions
+# [context-and-expression-syntax]: https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions
+name: In-tree build & tests
+
+on:
+  push:
+    branches:
+      - master
+      - llvm_release_*
+    paths-ignore: # no need to check build for documentation changes
+      - 'docs/**'
+      - '**.md'
+  pull_request:
+    branches:
+      - master
+      - llvm_release_*
+    paths-ignore: # no need to check build for documentation changes
+      - 'docs/**'
+      - '**.md'
+  schedule:
+    # Ideally, we might want to simplify our regular nightly build as we
+    # probably don't need every configuration to be built every day: most of
+    # them are only necessary in pre-commits to avoid breakages
+    - cron: 0 0 * * *
+
+env:
+  LLVM_VERSION: 12
+
+jobs:
+  build_and_test_linux:
+    name: Linux
+    strategy:
+      matrix:
+        build_type: [Release, Debug]
+        shared_libs: [NoSharedLibs]
+        include:
+          - build_type: Release
+            shared_libs: EnableSharedLibs
```

Comment:
Could you clarify why we don't test debug configuration with enabled shared libs?

Reply:
We can easily enable those - I just though that it is not that important to have such configuration